### ISSUE TITLE
Change SickRage to new URL

### ISSUE THREAD

**Issue by mathewcalvin** [SynoCommunity/spksrc#1989]
The sickrage team has left the "SICKRAGETV" git group and formed a new one (just "SickRage") after a sort of revolt/upheaval in leadership.  The only reason this matters on this forum is that Sickrage is broken on new installs for Synology devices with the old URL. (Python eats all system memory and crashes repeatedly when adding shows)

Here's the new URL: git://github.com/SickRage/SickRage.git

There are a lot of new features being added, which is nice.

**Comment by Diaoul** [comment]
@Dr-Bean: I think we can remove the package and redirect users to SickBeard Custom?

**Comment by Dr-Bean** [comment]
SB-Custom needs some work. It should be on someone's TODO list...yours? ;)
- Change default repo to the correct one in wizard and installer;
- Add a recursive switch to the git clone, maybe add depth of 1 or so, and move the clone command inside an if-synopkg-install-statement;
- Drop the autoprocessTV permissions changes or change to 644 at the very least.

Besides that, it would be nice to inform the old and new SR group to update their HowTo's/Wiki's to use the correct package once the above is done.

**Comment by cytec** [comment]
for the records: https://github.com/SynoCommunity/spksrc/pull/1948

**Comment by miigotu** [comment]
Will be glad to update our documentation once this is ready.

As for the changes:
autoProcessTV no longer exists in either repo
python 2.7.9 or 2.7.10 is preferable if possible, as it removes the need for pyOpenSSL, otherwise a working pyOpenSSL is needed.
There are no other requirements =P

**Comment by cytec** [comment]
@miigotu afaik the current python version provided by SynoCommunity is 2.7.10 
@SynoCommunity/developers personally i'd keep the package as it is, as there are several changes/differences between SickBeard and SickRage when it comes to PP and stuff (like discontinued autoProcessTV etc)

**Comment by mathewcalvin** [comment]
@cytec I am running just fine on 2.7.10-14 (the synocommunity distro)
However I'd argue against keeping the package as is - it is broken for fresh synology installs on the SickRageTV github fork.  Since all the contributors except for one have moved to /sickrage/ (and since that's the fork that works for synology products at the moment) I'd argue that we should either swap to that fork or go to a blank "Sick*" package and offer users links to the different forks to choose from (sickbead-custom, sickrageTV, and SickRage)

**Comment by miigotu** [comment]
Whatever solution works I'll be happy, as long as users can easily use the updated source that actually works.

If the issues with sickbeard-custom are fixed it shouldnt be a problem, but ssl verification currently doesnt work properly with it.

**Comment by Dr-Bean** [comment]
That's what the recursive switch is for, @cytec. PPscripts were added in as submodules. Happens to work for either SR project, and it doesn't matter for forks that don't use submodules: the recursive switch won't do anything.

@miigotu SSL verification should not be an issue afaik? Our Python package can handle anything you throw at it, and we have a number of Python-based packages with SSL capabilities...and no issues. 
The only exception would be self-signed certificates with indexers etc, but that's expected.

**Comment by cytec** [comment]
@mathewcalvin well... we already had that discussion in the PR i mentioned above... so no need to go through all of that again... one can change the URL on install Time to what ever repo he/she likes.

**Comment by miigotu** [comment]
Ill agree that Im not a fan of the "Beard" part since more code has been rewritten than SB ever had, and users stumbling onto it might not realize it houses SR also, but thats sort of the least of our worries.

@Dr-Bean Im not sure exaclt what the cause was as there have been several issues across different platforms with SSL SNI and also separately with sutes using newer transports such as TLS1.2.

I do know that there were problems with the sickbeard-custom package with this.
@neoatomic could you elaborate for me here?

**Comment by neoatomic** [comment]
I'm in agreement with @Cytec and @miigotu on this.

Upgrading/modifying the current Sickrage package seems the smartest thing to do. As all Sickrage users have the Sickrage package installed/running.
We dont want a mass migration of users from the Sickrage package to the Sickbeard-custom package. As it really doesn't solve anything, and support wise that would be a nightmare for everyone...


Also focusing on the Sickrage package will have some additional benefits. : 

* The SickRage team plans to implement some major new features in the future. If those requere a change in the package, than this can easily be done with a `own` maintained package. 
(The same might not be possible with the sickbeard-custom package as functionality needs to be universal for all sickbeard forks.)
* Another benefit could be bumping the version number of the package to easyly migrate users to the new Repo/URL. 

I will offer my time for Wiki, How-To's and support related issues if needed.

ps. Development/support for sickbeard has long stopped, so would make more sense to move away support from sickbeard to SickRage.

@miigotu The Python package is a separate package and installed stand-alone, so other Synocommunity Apps can use it and not only SR.

**Comment by Dr-Bean** [comment]
Ok. Simply put, the SB-custom package needs an update, because even though the development on the original SB has stopped, there are forks out there that are still maintained...including SR. We will continue maintaining the package as long as that need exists.
There are plenty of people running SR via SB-Custom, and I don't see a reason why that shouldn't continue at this point.
Note that the two packages (SR and SB-custom) are functionally identical. The only difference is the name, and that discussion does not belong here. If there are major changes in SR's codebase that necessitate further changes, we'll be happy to hear about it.

**Comment by neoatomic** [comment]
@Dr-Bean Was just pointing out that the user base for sickbeard is getting very small, so in my view it would be logical to focus more on the SickRage requirements. NOT suggesting sb-custom should be removed off-course. :) 
The user-base of sickbeard forks is about 4% (unique ip's) and SickRage is about 35%, according to a popular NZB indexer.  
As the SR installs on the SB-custom installer concerns, its a very low percentage. Rarely i see a log that still uses the SB-custom installer. 

So again we should focus on improving/modifying the SickRage package that the vast majority of users is currently using.
Migrating them to sb-custom doesn't solve anything, on the counterpart it will only create lots of support issues, and its less flexible to the future.

Improvements for the package: 

* Allow Git clone to add submodules. (for nzbtomedia)
* Change URL to the new Repo.
* Set local on startup? (currently detection not working, as it shows none/none)

**Comment by Dr-Bean** [comment]
@Diaoul The SB-custom package has been updated.

**Comment by Dr-Bean** [comment]
Closing. As stated previously, the URL will not be changed.

**Comment by miigotu** [comment]
That wasn't what @cytec and I discussed. Also, if sickbeard-custom is to be used, then the sickrage package needs removed.

SickRage/SickRage is the main/community version of the original SRTV. Being as this is a community repo wouldnt it make sense to include the WORKING community version of an app?

@Dr-Bean really seems you have some bias here.

**Comment by Dr-Bean** [comment]
Let's keep the personal attacks, along with the spreading of FUD out of here please. I have no idea what you and @cytec discussed. I responded to the person who opened the issue, not to you.

Since we're having the discussion then: It seems you prefer the SB-Custom package over the SR package? Sure. If you want time to update your manual, let me know how long you need. Otherwise, we can turn off distribution immediately.

**Comment by Diaoul** [comment]
Both original SR and SR "community" can be installed from the SB-custom package. As far as I'm concerned you have a way to install *your* SR on a Synology NAS. So what is exactly the request here? Change the existing "SR" package to point to the "community" one? Remove it?

It's unlikely we will do such things, for the sake of existing users that chose to install the original one. If they want to switch or just install *yours*, they are free to do it using the SB-custom package as you can instruct them to do.

**Comment by Dr-Bean** [comment]
@Diaoul Or your option. Either works ;)

**Comment by miigotu** [comment]
I guess Ill just make a package too then. If SRTV can have their own special little package (who has less than 200 users now) but we can't (>6800 users) that isnt very fair now is it?

**Comment by Dr-Bean** [comment]
Yep, that should work fine. Fork spksrc, you can use the existing code and adjust to your needs.

**Comment by Diaoul** [comment]
Don't get me wrong, if at some point the original SR package becomes unmaintained/unpopular, we will remove it.
From a maintainer perspective, this is a nightmare. Forking without changing name is a real pain to maintain. Maybe when things cooldown we will reconsider but for now the SB-custom is a way to install *your* SR easily.

**Comment by cytec** [comment]
that's what @miigotu and i discussed. i told him more or less what @Diaoul said in his last post: We wont change the url in the near future but i will keep an eye on activity on the "old" and "new" SickRage and that we may change the url if the "old" one doesn't get updates for a while. in the meantime the users can still just change the url ...

**Comment by Dr-Bean** [comment]
Well, let's add the recursive switch and either copy the autoprocessTV changes from SB-custom/drop those lines altogether for the SR package then. That way people can still choose which package to use, and it won't make a difference. @cytec, care to do that?

**Comment by neoatomic** [comment]
@Diaoul I actually tent to agree with you. I also see the difficulties from your position.

However i have a feeling none of you are actually using SickRage.? I would suggest simply trying to install Sickrage from the old repo.
You will then discover that it will not even start, its that badly broken... And i'm not even talking about the crippling bugs that there are if it was running.  Again, just try it for yourself and form your own opinion.

And as @miigotu has mentioned 80% of all the user numbers that we had on the old repo have moved to the new one. The 20% that did not move are probbebly stuck there due to outdated installers or uninstalled/moved to Sonar or others. That in-itself is pretty telling and convincing number. 
Also pretty much all other OS installers have moved/changed to the new Repo. 

In Summery, its clearly not in the best interest of the users to let the package point to a non working Repository...

**Comment by miigotu** [comment]
Just so it doesn't seem like I'm making these numbers up, here is a screenshot from a 2 week period:
![image](https://cloud.githubusercontent.com/assets/390379/12428801/9ee35a5a-be9b-11e5-82c6-55bfad163eb4.png)

**Comment by cytec** [comment]
@neoatomic @miigotu personally i use SickRage but as is said before in IRC im useing my own fork so i don't have any install problems, but there is https://github.com/SynoCommunity/spksrc/issues/2087#issuecomment-173130523 which is exactly what i was expecting to happen sooner or later (with or without the url change)... 

There happens exactly what @neoatomic is talking about: upstream changed dependencie handling like he doesn't care about backwards compatibility... solution would be to create wheels for those requirements and install them. 

Personally i'm likely to support this request now, as there are problems with upstream and we have to "fix" the package anyway. so changing the URL wouldn't make a big difference... only "problem" i see at the moment is:

1. How to migrate already installed versions to the new url without the user having to do anything
2. Is SickRage compatible to SickRageTV? Database etc?

\cc @Dr-Bean

**Comment by miigotu** [comment]
Yes, he is getting even worse. 736 people switched to our fork yesterday alone since he alienated over 75% of his remaining users with the pip requirement who cannot easily get-pip (and those who CAN have pip/distutils he tells them to run as root? -.-)

Config/Databases are compatible.

I would assume that a higher package version which switches the url in the remote and git resets --hard origin is enough.

For the most part what we instruct users to do is:
git remote set-url origin https://github.com/SickRage/SickRage.git
git fetch origin
git reset --hard origin/master (regardless of the branch they are on, updater will be able to update them from there)

And I usually throw in a `git prune` and `git remote prune origin` to try and avoid any non-existent refs

![image](https://cloud.githubusercontent.com/assets/390379/12445037/e5305aaa-bf16-11e5-81ef-8c639f7628dc.png)

**Comment by Diaoul** [comment]
See [my previous comment](https://github.com/SynoCommunity/spksrc/issues/1989#issuecomment-171370692) on this. Let's remove the SickRage package from the repo for now and redirect users to SickBeard-Custom. Install should work just fine from there.
Installing the "original" SickRage won't work with our SB-Custom but I don't mind that as it's gotten away from an SB fork anyway and changed the way requirements are managed without notice.

**Comment by Dr-Bean** [comment]
Deactivated. No sense in keeping a broken package active.

**Comment by Dr-Bean** [comment]
@miigotu / @neoatomic: I assume either of you can take care of changing the FAQ here: https://github.com/SickRage/sickrage-issues/wiki/Switching-your-Synology's-Sickrage-to-the-new-repository#install-sickrage?

**Comment by miigotu** [comment]
So it shouldnt be a problem if I PR a new package then?

I don't see how denying our unique identity and meld it with the even older non-maintained original (sick-beard) that we share very little code with makes this any simpler or clearer to anyone. It just isn't good for anyone.

The team, the community, and all other 3rd parties we have went to have had no issue seeing this as a simple switch. And that was before he went and made it worse over the past few weeks.

We didn't "steal" his project or "his" name or anything like that. He came back and practically destroyed the project by booting the entire team. In anticipation of his further destruction, we SAVED those people from his craziness, and kept the real project alive. I mean, it wasnt like we were actively advertising it and trying to con users into switching or something. He subsequently actually did destroy the project, luckily we were there to pick up the shrapnel.

It seems silly not to err on the side of tens of thousands of users, who chose on their own, I mean, from a maintaners' standpoint and all.

**Comment by neoatomic** [comment]
The thing here is that we prefer to have a dedicated package for our users. 
Main reasons for that are : 

* Allows the possibility/flexibility to implement package changes in the future. 
* This will give our users ease of installation.

With that large of a user-base i think its not to much to ask for. 
And as there is already a dedicated Sickrage package i dont see the need to confuse users even more by letting (jet again...) switch to another package.
Its just not practical, and a nightmare support wise. Its not just writing a new wiki, also paths/ports change for users and cache might act up if restored etc.
Therefor i would be in favor of fixing-up the current sickrage package, to prevent any issues/limitations in the future. Clearly best to solve it all in one go and make it future proof.
We have a lot of ideas/thoughts how we want to go forward with the project, and functionality/code will change substantially.
And if the name is confusing than just call it Sickrage Ce or whatever is preferred.

Also migrating the remaining users from the old repo isn't really necessary in my view, the number that is still stuck there is low and not really worth the effort. But if that is the way you want to go, i think that wont be to hard to do.

Sickbeard is basically death and unmaintained, as are the sickbeard forks. All community activity surrounds Sickrage. 
Yes Sickrage uses a similar installation method, but so do coutchpatato & coutchpatato custom and other packages. 
So i find it unfair to use that as it as an excuse to redirect users there. 
Especially when Miigotu has offered to maintain the package in the future, and so ensuring functionality/availability and saving you all that work.  :)

Anyway, on a side-note, what is the status of App based user accounts that Synology planned/wanted to implement in DSM.? Is that still in the pipeline?

**Comment by Diaoul** [comment]
Everything has being said on that topic. Unless you have new elements to add please restrain for commenting.
* You have a way for your users to install SickRage "community" with a Package distributed and supported by SynoCommunity. As far as packaging goes the job is done here.
* If you think that the name of the package is confusing, feel free to rename your app. The fact that you named your fork the same name as the original is not SynoCommunity's concern.

We'll eventually revisit this issue in a few months if it is proven that the original repo is dead and the "community" one its rightfull successor.

Sick* user accounts based on DSM authentication is not something we're not working on. Feel free to work on that.

**Comment by neoatomic** [comment]
@Diaoul Sadly you seem to interpreter my text different/negatively, that clearly wasn't my intention...
Apologies for that. I will leave the talking to @miigotu then.

@cytec May i ask what you will be advising your users/community.? To use the sb-custom, or will you host the package again on your own SPK server.?

**Comment by Rio-D** [comment]
Can someone please update me on what is happening here?

I noticed the sick rage app was removed and everything Sickrage related has stopped working. 

Based on reading the above my understanding is, there was a breakup and there is a new sick rage called 'Sickrage' that is working?

**Comment by Dr-Bean** [comment]
@Rio-D Please see https://github.com/SynoCommunity/spksrc/issues/2087#issuecomment-173152381, that's a better fit for your question.

**Comment by cytec** [comment]
@neoatomic personally i don't care... i've made the SickRage package only because i've also have a SickBeard fork, which some ppl installed with SB-custom and then wanted to try my SR fork too. So as an easy solution – without deleting/overwriting a working SB – i've created a seperate SPK for ONLY that reason... 

As time passed by more and more ppl where useing it and i thought i might do a PR for it and putt it here. If the current Package isn't working anymore i have no problems with changing it, and host it on my repo again.

**Comment by reyman** [comment]
I try the git url `git://github.com/SickRage/SickRage.git` given in this post with sickboard custom, but sickrage failed to launch on ds213+.

I search where can i find the log of sickrage launch, but i found some error logged during install : 

```2016/01/22 15:09:53     install  :
                        Begin:  /bin/tar xf /volume2/@tmp/@synopkg/@download/sickbeard-custom/@SYNOPKG_DOWNLOAD_sickbeard-custom -C /volume1/@tmp/pkginstall conf --no-same-owner
/bin/tar: conf: Not found in archive
/bin/tar: Exiting with failure status due to previous errors
```

**Comment by Dr-Bean** [comment]
@reyman Both messages will not stop the package from operating. As there's no relationship with this particular issue, please open a new issue, so that we can properly troubleshoot.
Before you do, please check [CONTRIBUTING](https://github.com/SynoCommunity/spksrc/blob/master/CONTRIBUTING.md#issues).

**Comment by JackDandy** [comment]
Hi @Diaoul @cytec @KaraokeStu

Thank you for maintaining the SickBeard Custom package etc. to give users choice on which fork they would prefer to use.  A particular comment in this thread made me raise an eyebrow...

> Sickbeard is basically death and unmaintained, as are the sickbeard forks. All community activity surrounds Sickrage

This is incorrect and could be read as an attempt to subvert user choice to one fork.  SickGear is a year older than the sickrage fork discussed here and has always linked [SickBeard Custom package installs SickGear](https://github.com/SickGear/SickGear.Wiki/wiki/Install-SickGear-%5B5%5D-Synology)

Proud of our heritage, we always maintained Sick-Beard author credits.  Also, we have never changed the Sick-Beard API or removed the autoProcessTV system that users from Sick-Beard to SickGear use to this very day.  I don't know if you did remove your installer support for autoProcessTV, but if you did then that may have broken forks that remained close to the original Sick-Beard.  That said, I haven't had any complaints from our Syno 5 or 6 users.  

I just want to add color to this thread so that people can see that choices are available thanks to this package.

> We will continue maintaining the package as long as that need exists.

Again, thank you for making our life on project SickGear simpler by maintaining this package -- especially during times where it's clear that you have been above the attempts to stifle the legacy of Sick-Beard, its forks, and the fact that choice is available.

Now we wait for this [DMCA abuse](https://github.com/SickGear/SickGear.Wiki/wiki) to be lifted so that we can resume as we were.

**Comment by miigotu** [comment]
This issue and related PR's were concerned with SickRage packages, not the sickbeard-custom package.  I feel like you just wanted to plug your repo because it's coming up a day sooner than ours. GL with that.

**Comment by KaraokeStu** [comment]
I'm locking this issue
